File: planeGame_android_kotlin/app/src/main/java/cn/lblbc/game/sprite/EnemyPlane.java
/**
 * 厦门大学计算机专业 | 前华为工程师
 * 专注《零基础学编程系列》  http://lblbc.cn/blog
 * 包含：Java | 安卓 | 前端 | Flutter | iOS | 小程序 | 鸿蒙
 * 公众号：蓝不蓝编程
 */
package cn.lblbc.game.sprite;

import android.graphics.Bitmap;

import cn.lblbc.game.SpriteManager;

/**
 * 敌机
 */
public class EnemyPlane extends AutoSprite {

    public EnemyPlane(Bitmap bitmap) {
        super(bitmap);
    }

    public void hide() {
        explode();
        super.hide();
    }

    //创建爆炸效果
    private void explode() {
        float centerX = getX() + getWidth() / 2;
        float centerY = getY() + getHeight() / 2;
        SpriteManager.getInstance().addExplosion(centerX, centerY);
    }
}

File: planeGame_android_kotlin/app/src/main/java/cn/lblbc/game/sprite/EnemyPlane.kt
/**
 * 厦门大学计算机专业 | 前华为工程师
 * 专注《零基础学编程系列》  http://lblbc.cn/blog
 * 包含：Java | 安卓 | 前端 | Flutter | iOS | 小程序 | 鸿蒙
 * 公众号：蓝不蓝编程
 */
package cn.lblbc.game.sprite

import android.graphics.Bitmap
import cn.lblbc.game.SoundManager
import cn.lblbc.game.SpriteManager

/**
 * 敌机
 */
class EnemyPlane(bitmap: Bitmap) : AutoSprite(bitmap) {
    override fun hide() {
        explode()
        super.hide()
    }

    //创建爆炸效果
    private fun explode() {
        val centerX = x + width / 2
        val centerY = y + height / 2
        SoundManager.getInstance().playBomb()
        SpriteManager.addExplosion(centerX, centerY)
    }
}